cpp/src/slice2cpp/Gen.cpp

```diff
@@ -2311,7 +2038,7 @@ Slice::Gen::ProxyVisitor::visitOperation(const OperationPtr& p)
         //
         // Generate a read method if there are more than one ret/out parameter. If there's
         // only one, we rely on the default read method from LambdaOutgoing
-        // except if the unique ret/out is optional or is an array/range.
+        // except if the unique ret/out is optional or is an array.
```

Comment:
This comment is actually not correct. The array metadata doesn't affect the "promise" mapping.